syntax = "proto3"; 

package proto_tc;

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates cord = 2;
    uint32 id = 3;
}

message Bus {
    string name = 1;
    bool is_circular = 2;
    repeated uint32 id_stop = 3;
}

message DistanceBetweenStops {
    int32 id_from = 1;
    int32 id_to = 2;
    int32 distance = 3;
}

message TransportCatalogue {
    repeated Stop stop = 1;
    repeated Bus bus = 2;
    repeated DistanceBetweenStops distance = 3;
}